src/datascience-ui/data-explorer/mainPanel.tsx

```diff
@@ -304,17 +309,60 @@ export class MainPanel extends React.Component<IMainPanelProps, IMainPanelState>
         return [];
     }
 
-    private normalizeRows(rows: JSONArray): ISlickRow[] {
+    private normalizeData(rows: JSONArray): ISlickRow[] {
+        // While processing rows we may encounter Inf, -Inf or NaN.
+        // These rows' column types will initially be 'string' or 'object' so
+        // make sure we update the column types
+        // Set of columns to update based on this batch of rows
+        const columnsToUpdate = new Set<string>();
```

Comment:
`normalizeRows` already loops through the row data, so I implemented checking cell data for inf/-inf/nan inside the loop. I thought about whether to pull the column checks out into a separate function but didn't really want to because we'd either need to maintain some global state and check it after this function returns, or return the set of columns to update from this function. So ended up renaming this function to `normalizeData` as that's what it's doing. But I'm open to feedback on whether this is confusing.

Reply:
Outside of strict statistics 'normalize' is pretty overloaded. To me at least this type of operation doesn't seem too out of place to be happening here..